nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/main/java/org/apache/nifi/schema/access/SchemaAccessUtils.java

```diff
@@ -85,7 +85,7 @@
             .description("Specifies the name of the branch to use when looking up the schema in the Schema Registry property. " +
                     "If the chosen Schema Registry does not support branching, this value will be ignored.")
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
-            .expressionLanguageSupported(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
```

Comment:
Can't SCHEMA_BRANCH_NAME and SCHEMA_VERSION possibly evaluate EL using FlowFile attributes like SCHEMA_NAME (from [SchemaNamePropertyStrategy](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-standard-record-utils/src/main/java/org/apache/nifi/schema/access/SchemaNamePropertyStrategy.java#L62))? I traced the calls back to [RecordReaderFactory](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-service-api/src/main/java/org/apache/nifi/serialization/RecordReaderFactory.java#L46), where you _might_ have FlowFile attributes available.

Reply:
Ok. Looks like you took care of that in 5145? So I rebased and pushed a new copy. Looks clear to me now.